src/wolfio.c

```diff
@@ -473,11 +473,28 @@ int EmbedReceiveFrom(WOLFSSL *ssl, char *buf, int sz, void *ctx)
     recvd = TranslateReturnCode(recvd, sd);
 
     if (recvd < 0) {
+        int err;
         WOLFSSL_MSG("Embed Receive From error");
-        recvd = TranslateIoError(recvd);
-        if (recvd == WOLFSSL_CBIO_ERR_WANT_READ &&
+
+        err = TranslateIoError(recvd);
+        if ((err == WOLFSSL_CBIO_ERR_WANT_READ ||
+                err == WOLFSSL_CBIO_ERR_ISR) &&
             !wolfSSL_dtls_get_using_nonblock(ssl)) {
-            recvd = WOLFSSL_CBIO_ERR_TIMEOUT;
+            err = WOLFSSL_CBIO_ERR_TIMEOUT;
+        }
+        return err;
+    }
+    else if (recvd == 0) {
+        int type;
+        XSOCKLENT length = sizeof(int); /* optvalue 'type' is of size int */
+
+        if (getsockopt(sd, SOL_SOCKET, SO_TYPE, &type, &length) == 0 &&
```

Comment:
Now that we use this code twice it may be worthwhile to do a small static function to get the socket type.

Reply:
Refactored in one func.